src/test/java/teammates/logic/core/FeedbackResponsesLogicTest.java

```diff
@@ -1076,6 +1077,69 @@ public void testGetSessionResultsForCourse_allQuestionsSpecificSection_shouldGen
         assertEquals(0, commentGiverVisibilityTable.size());
     }
 
+    @Test
+    public void testGetSessionResultsForCourse_responseFetchByGiverOrReceiverOnly_shouldGenerateCorrectBundle() {
+        var responseBundle = loadDataBundle("/FeedbackSessionResultsTest.json");
+        removeAndRestoreDataBundle(responseBundle);
+
+        var session = responseBundle.feedbackSessions.get("standard.session");
+
+        var instructor = responseBundle.instructors.get("instructor1OfCourse1");
+        var sectionToTest = "Section A";
+        var questionResponseMapByGiver = frLogic.getSessionResultsForCourse(
+                session.getFeedbackSessionName(), session.getCourseId(), instructor.getEmail(),
+                null, sectionToTest, FeedbackResultFetchType.GIVER)
+                .getQuestionResponseMap();
+        for (var questionResponse : questionResponseMapByGiver.entrySet()) {
+            var responses = questionResponse.getValue();
+            responses.forEach(resp -> {
+                assertEquals(sectionToTest, resp.getGiverSection());
+            });
+        }
+
+        var questionResponseMapByReceiver = frLogic.getSessionResultsForCourse(
+                session.getFeedbackSessionName(), session.getCourseId(), instructor.getEmail(),
+                null, sectionToTest, FeedbackResultFetchType.RECEIVER)
+                .getQuestionResponseMap();
+        for (var questionResponse : questionResponseMapByReceiver.entrySet()) {
+            var responses = questionResponse.getValue();
+            responses.forEach(resp -> {
+                assertEquals(sectionToTest, resp.getGiverSection());
+            });
+        }
+    }
+
+    @Test
+    public void testGetSessionResultsForCourse_splitResponseFetchByGiverAndReceiver_shouldGenerateCorrectBundle() {
+        var responseBundle = loadDataBundle("/FeedbackSessionResultsTest.json");
+        removeAndRestoreDataBundle(responseBundle);
+
+        var session = responseBundle.feedbackSessions.get("standard.session");
+
+        var instructor = responseBundle.instructors.get("instructor1OfCourse1");
+        var questionResponseMapFromMultiFetch = frLogic.getSessionResultsForCourse(
```

Comment:
Try to use `var`s more discreetly; it comes at the obvious expense of code readability (for e.g., what variable type is this?). Yes, IDEs can tell you that e.g. by just hovering over, but if you have to rely on IDE to know the variable type, something is not right.

Reply:
I see. I do agree that sometimes the type should be explicit.

It is perhaps that I see type inference in newer languages (Go, Rust, Kotlin) and therefore tend to use it everywhere. I suppose if the method name already carries the hint on the type, I will use `var` instead; otherwise, a type is better.